orte/mca/ess/base/ess_base_std_prolog.c

```diff
@@ -56,6 +59,10 @@ int orte_ess_base_std_prolog(void)
             goto error;
         }
     }
+    /* check for OPAL_PREFIX */
+    if (NULL != (evar = getenv("OPAL_PREFIX"))) {
+        opal_setenv("PMIX_PREFIX", evar, false, &environ);
```

Comment:
The second case will indeed be propagated. Today, mpirun harvests all envars starting with OMPI_ and PMIX_ and forwards them. If someone direct launches, then the launcher is responsible for forwarding the environment. Thus, if someone sets PMIX_PREFIX in their environment, it should get to the backend. There is no way we can determine whether or not PMIX_PREFIX is required for the external PMIx installation.

So the only case we have to deal with is the one where OPAL_PREFIX is set and we need to ensure that PMIX_PREFIX corresponds. As you correctly pointed out, this only happens when we are using the embedded version of PMIx.